app/components/candidate_interface/course_choices_review_component.rb

```diff
@@ -158,6 +158,7 @@ def start_date_row(application_choice)
     end
 
     def degree_required_row(application_choice)
+      return if application_choice.current_course.undergraduate?
```

Comment:
I added another return clause as would be my preference to be more explicitly.

If you strongly prefer to add `&& !application_choice.current_course.undergraduate?` below, let me know.

Reply:
Could we move both `application_choice.current_course.undergraduate?` and `application_choice.current_course.degree_grade?` into a private method 

> `course_does_not_require_degree?(application_choice)`

then just return if that new method is true. That would keep this clean.

Reply:
Actually you gave me a good idea. I move this to the course model as it makes more sense to be there. I moved there and I think now everything is on the right place.